packages/mask/src/social-network-adaptor/mirror.xyz/injection/PostActions/index.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+    createInjectHooksRenderer,
+    PostInfo,
+    PostInfoProvider,
+    useActivatedPluginsSNSAdaptor,
+    usePostInfoDetails,
+} from '@masknet/plugin-infra/content-script'
+import { Plugin } from '@masknet/plugin-infra'
+import { Flags } from '../../../../../shared/index.js'
+import { createReactRootShadowed } from '../../../../utils'
+import { noop } from 'lodash-unified'
+import { formatEthereumAddress } from '@masknet/web3-shared-evm'
+
+const ActionsRenderer = createInjectHooksRenderer(
+    useActivatedPluginsSNSAdaptor.visibility.useNotMinimalMode,
+    (plugin) => plugin.TipsRealm?.UI?.Content,
+)
+
+export function PostActions({ isFocusing }: { isFocusing?: boolean }) {
+    const identifier = usePostInfoDetails.author()
+    const nickname = usePostInfoDetails.nickname()
+    const coAuthors = usePostInfoDetails.coAuthors()
+
+    if (!identifier) return null
+    return (
+        <ActionsRenderer
+            // In Mirror, then profile identifier is wallet address
+            tipsAccounts={[
+                {
+                    address: identifier.userId,
+                    name: nickname ? `(${nickname}) ${formatEthereumAddress(identifier.userId, 4)}` : identifier.userId,
+                },
+                ...(coAuthors?.map((x) => ({
+                    address: x.author.userId,
+                    name: x.nickname ? `(${x.nickname}) ${formatEthereumAddress(x.author.userId, 4)}` : x.author.userId,
+                })) ?? []),
+            ]}
+            identity={identifier}
+            slot={Plugin.SNSAdaptor.TipsSlot.MirrorEntry}
+        />
+    )
+}
+
+function createPostActionsInjector() {
+    return function injectPostActions(postInfo: PostInfo, signal: AbortSignal) {
+        const jsx = (
+            <PostInfoProvider post={postInfo}>
+                <PostActions />
+            </PostInfoProvider>
+        )
+        if (postInfo.actionsElement) {
+            const root = createReactRootShadowed(postInfo.actionsElement.afterShadow, {
+                key: 'post-actions',
+                signal,
+            })
+            if (postInfo.actionsElement?.realCurrent?.parentNode?.lastElementChild) {
+                ;(postInfo.actionsElement.realCurrent.parentNode.lastElementChild as HTMLDivElement).style.flex = '1'
+                ;(postInfo.actionsElement.realCurrent.parentNode as HTMLDivElement).style.flex = '1 1 auto'
```

Comment:
I don't think it's necessary to force casting as `HTMLDivElement` here.

Reply:
The `postInfo.actionsElement.realCurrent.parentNode` type is ParentNode, don't have `style` property.